org.mixedrealitytoolkit.spatialmanipulation/ObjectManipulator/ObjectManipulator.cs

```diff
@@ -811,15 +849,15 @@ private void ApplyTargetTransform()
             TransformFlags modifiedTransformFlags = TransformFlags.None;
             ModifyTargetPose(ref targetTransform, ref modifiedTransformFlags);
 
-            if (rigidBody == null)
+            if (rigidBody == null || (rigidBody.isKinematic && currentRigidbodyMovementType == MovementType.Instantaneous))
             {
                 HostTransform.SetPositionAndRotation(targetTransform.Position, targetTransform.Rotation);
                 HostTransform.localScale = targetTransform.Scale;
             }
             else
             {
                 // There is a Rigidbody. Potential different paths for near vs far manipulation
-                if (!useForces)
+                if (!UseForces)
                 {
                     rigidBody.MovePosition(targetTransform.Position);
                     rigidBody.MoveRotation(targetTransform.Rotation);
```

Comment:
Note:  I left these original rigiBody.MovePosition/Rotation here in ApplyTargetTransform() which operates in update loop, though technically I think you're supposed to do these in the Fixed update loop.